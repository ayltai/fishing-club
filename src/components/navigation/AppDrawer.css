.drawer {
    top   : 64px !important;
    width : 230px !important;
}
