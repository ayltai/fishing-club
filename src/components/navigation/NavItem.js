// @flow
'use strict';

import { extendObservable } from 'mobx';

export default class NavItem {
    key  : string;
    name : ?string;
    icon : ?string;
    page : any;

    constructor(key : string, name : ?string, icon : ?string, page : any) {
        extendObservable(this, {
            key  : key,
            name : name,
            icon : icon,
            page : page
        });
    }
}
