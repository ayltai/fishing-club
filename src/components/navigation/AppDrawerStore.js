// @flow
'use strict';

import { extendObservable } from 'mobx';
import NavItem from './NavItem';
import Rx from 'rx-lite';

export default class AppDrawerStore {
    isOpened         : boolean;
    selected         : string;
    items            : NavItem[];
    selectionChanges : Rx.Subject;

    constructor(isOpened : boolean = true) {
        extendObservable(this, {
            isOpened : isOpened,
            selected : '',
            items    : []
        });

        this.selectionChanges = new Rx.Subject();
    }
}
