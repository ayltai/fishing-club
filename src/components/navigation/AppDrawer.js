// @flow
'use strict';

import React from 'react';
import PropTypes from 'prop-types';
import { observer } from 'mobx-react';
import muiThemeable from 'material-ui/styles/muiThemeable';
import Divider from 'material-ui/Divider';
import Drawer from 'material-ui/Drawer';
import FontIcon from 'material-ui/FontIcon';
import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
import AppDrawerStore from './AppDrawerStore';
import NavItem from './NavItem';
import './AppDrawer.css';

@observer
class AppDrawer extends React.Component {
    static propTypes : Object;

    props : {
        store : AppDrawerStore
    };

    render() : any {
        return (
            <Drawer
                docked={this.props.store.isOpened}
                containerClassName="drawer">
                <Menu value={this.props.store.selected}>
                    {this.props.store.items.map((item : NavItem, index : number) : any => {
                        if (item.key === '-') return <Divider key={`${item.key}-${index}`} />;

                        const icon : any = item.icon ? <FontIcon className={item.icon} /> : null;

                        return (
                            <MenuItem
                                key={`${item.key}-${index}`}
                                value={item.name}
                                primaryText={item.name}
                                leftIcon={icon}
                                onTouchTap={() : void => {
                                    this.props.store.selected = item.key;

                                    this.props.store.selectionChanges.onNext(item.key);
                                }} />
                        );
                    })}
                </Menu>
            </Drawer>
        );
    }
}

AppDrawer.propTypes = {
    store : PropTypes.instanceOf(AppDrawerStore)
};

export default muiThemeable()(AppDrawer);
