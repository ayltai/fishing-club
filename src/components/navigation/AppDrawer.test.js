'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import lightBaseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import AppDrawer from './AppDrawer';
import AppDrawerStore from './AppDrawerStore';
import NavItem from './NavItem';

const createStore : AppDrawerStore = () => {
    const store : AppDrawerStore = new AppDrawerStore(true);

    store.items.push(new NavItem('/', 'Home', 'home'));
    store.items.push(new NavItem('/location', 'Location', 'map-marker'));
    store.items.push(new NavItem('/transportation', 'Transportation', 'bus'));

    store.selected = '/';

    return store;
};

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <MuiThemeProvider muiTheme={getMuiTheme(lightBaseTheme)}>
            <AppDrawer store={createStore()} />
        </MuiThemeProvider>,
        div);

    expect(div.innerHTML).toMatchSnapshot();
});
