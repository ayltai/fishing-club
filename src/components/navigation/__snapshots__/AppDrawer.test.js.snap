// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div data-reactroot>
  <!-- react-empty: 2 -->
  <div class="drawer"
       style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23); border-radius: 0px; height: 100%; width: 256px; position: fixed; z-index: 1300; left: 0px; top: 0px; overflow: auto;"
  >
    <div role="presentation"
         style="z-index: 1000; width: 84px;"
    >
      <div role="menu"
           style="padding: 8px 0px 8px 0px; display: table-cell; width: 84px;"
      >
        <div>
          <span role="menuitem"
                tabindex="0"
                style="border: 10px; box-sizing: border-box; display: block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: 16px; font-weight: inherit; position: relative; z-index: 1; color: rgba(0, 0, 0, 0.87); line-height: 48px; min-height: 48px; white-space: nowrap; background: none;"
          >
            <div>
              <div style="margin-left: 0px; padding: 0px 16px 0px 72px; position: relative;">
                <span class="home"
                      color="#757575"
                      style="color: rgb(117, 117, 117); position: absolute; font-size: 24px; display: block; height: 24px; width: 24px; top: 0px; margin: 12px; left: 4px;"
                >
                </span>
                <div>
                  Home
                </div>
              </div>
            </div>
          </span>
        </div>
        <div>
          <span role="menuitem"
                tabindex="0"
                style="border: 10px; box-sizing: border-box; display: block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: 16px; font-weight: inherit; position: relative; z-index: 1; color: rgba(0, 0, 0, 0.87); line-height: 48px; min-height: 48px; white-space: nowrap; background: none;"
          >
            <div>
              <div style="margin-left: 0px; padding: 0px 16px 0px 72px; position: relative;">
                <span class="map-marker"
                      color="#757575"
                      style="color: rgb(117, 117, 117); position: absolute; font-size: 24px; display: block; height: 24px; width: 24px; top: 0px; margin: 12px; left: 4px;"
                >
                </span>
                <div>
                  Location
                </div>
              </div>
            </div>
          </span>
        </div>
        <div>
          <span role="menuitem"
                tabindex="0"
                style="border: 10px; box-sizing: border-box; display: block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: 16px; font-weight: inherit; position: relative; z-index: 1; color: rgba(0, 0, 0, 0.87); line-height: 48px; min-height: 48px; white-space: nowrap; background: none;"
          >
            <div>
              <div style="margin-left: 0px; padding: 0px 16px 0px 72px; position: relative;">
                <span class="bus"
                      color="#757575"
                      style="color: rgb(117, 117, 117); position: absolute; font-size: 24px; display: block; height: 24px; width: 24px; top: 0px; margin: 12px; left: 4px;"
                >
                </span>
                <div>
                  Transportation
                </div>
              </div>
            </div>
          </span>
        </div>
      </div>
    </div>
  </div>
</div>

`;
